webserver/views/datasets.py

```diff
@@ -92,6 +93,41 @@ def view(id):
     )
 
 
+def get_json(id):
```

Comment:
you're not using this method twice any more, so I would move it back into the `view_json` function.

If we were going to keep this method I would give it a different name (`dataset_to_public_json`) and make it take the result of `get_dataset` as an argument instead of an id. It would also need documentation to say what it does.